packages/hash/integration/src/tests/integration.test.ts

```diff
@@ -1115,47 +1110,55 @@ describe("logged in user ", () => {
     expect(pageEntities[0].properties).toMatchObject({ tokens: [] });
     expect(pageEntities[1].properties).toMatchObject(textPropertiesA);
 
-    // getEntities gives back entities orderes by updated_at
-    const { entities } = await client.getEntities({
-      accountId: page.accountId,
-    });
+    // Check text entity A was created with the correct properties & outgoing links
 
-    expect(
-      entities
-        .slice(0, 5)
-        .map((entity) => ({
-          type: entity.entityTypeName,
-          props:
-            entity.entityTypeName === "Text" ? entity.properties : undefined,
-        }))
-        .sort((a, b) =>
-          // lexical sorting to circumvent concurrency of linked entity ordering when they are inserted
-          // Non-text blocks are first, page and block are not inserted concurrently and will be stable.
-          a.props === undefined || b.props === undefined
-            ? (a.props === undefined) !== (b.props === undefined)
-            : (a.props as any).tokens![0].text.localeCompare(
-                (b.props as any).tokens![0].text,
-              ),
-        ),
-    ).toEqual([
-      { type: "Page", props: undefined },
-      { type: "Block", props: undefined },
-      // The following 3 Text elements should be the concurrently inserted, nested entities
-      // sorting beforehand ensures reproducibility.
-      { type: "Text", props: textPropertiesA },
-      { type: "Text", props: textPropertiesB },
-      { type: "Text", props: textPropertiesC },
-    ]);
+    const textEntityA = (await Entity.getEntityLatestVersion(db, {
+      accountId: pageEntities[1].accountId,
+      entityId: pageEntities[1].entityId,
+    }))!;
+
+    expect(textEntityA).not.toBeNull();
+    expect(textEntityA.properties).toMatchObject(textPropertiesA);
+
+    const textEntityAOutgoingLinks = await textEntityA.getOutgoingLinks(db);
+
+    expect(textEntityAOutgoingLinks).toHaveLength(1);
+
+    const [textEntityAOutgoingLink] = textEntityAOutgoingLinks;
+
+    expect(textEntityAOutgoingLink.stringifiedPath).toBe("$.textB");
+
+    // Check text entity B was created with the correct properties & outgoing links
+
+    const textEntityB = await textEntityAOutgoingLink.getDestination(db);
+
+    expect(textEntityB.properties).toMatchObject(textPropertiesB);
+
+    const textEntityBOutgoingLinks = await textEntityB.getOutgoingLinks(db);
+
+    expect(textEntityBOutgoingLinks).toHaveLength(1);
+
+    const [textEntityBOutgoingLink] = textEntityBOutgoingLinks;
+
+    expect(textEntityBOutgoingLink.stringifiedPath).toBe("$.textC");
+
+    // Check text entity C was created with the correct properties & outgoing links
+
+    const textEntityC = await textEntityBOutgoingLink.getDestination(db);
+
+    expect(textEntityC.properties).toMatchObject(textPropertiesC);
+
+    const textEntityCOutgoingLinks = await textEntityC.getOutgoingLinks(db);
+
+    expect(textEntityCOutgoingLinks).toHaveLength(0);
```

Comment:
@thehabbos007 I've replaced this test so that we are no longer enforcing a particular order in which those links/entities are created/updated. Instead we check whether the relevant entities have been created with the correct `properties` and outgoing links. Let me know if I've missed something here.